Polyhedron/demo/Polyhedron/Plugins/Surface_mesh/Offset_meshing_plugin.cpp

```diff
@@ -117,58 +276,39 @@ CGAL::Three::Scene_item* cgal_off_meshing(QWidget*,
   CGAL::Timer timer;
   timer.start();
 
-  Offset_function offset_function(*tm_ptr, offset_value);
+  namespace p = CGAL::parameters;
 
-  Tr tr;
-  C2t3 c2t3 (tr);
+  Mesh_domain domain =
+    Mesh_domain::create_implicit_mesh_domain
+    (offset_function(tm_ptr, offset_value),
+     Sphere_3(center, sqrad),
+     p::relative_error_bound = 1e-7,
+     p::construct_surface_patch_index = [](int i, int j) { return (i * 1000 + j); });
 
-  // defining the surface
-  Surface_3 surface(offset_function,
-                    Sphere_3(center, sqrad)); // bounding sphere
+  CGAL::Mesh_facet_topology topology = CGAL::FACET_VERTICES_ON_SAME_SURFACE_PATCH;
+  if(tag == 1) topology = CGAL::Mesh_facet_topology(topology | CGAL::MANIFOLD_WITH_BOUNDARY);
+  if(tag == 2) topology = CGAL::Mesh_facet_topology(topology | CGAL::MANIFOLD);
+  Mesh_criteria criteria(p::facet_angle = angle,
+                         p::facet_size = sizing,
+                         p::facet_distance = approx,
+                         p::facet_topology = topology);
```

Comment:
This code is wrong (pointed by Sébastien). The manifold criterion, for Mesh_3, must be passed to `make_mesh_3`, and not to the `facet_topology` parameter of `Mesh_criteria`... And, thanks to unscoped enums, that is not detected at compile time.